### ISSUE TITLE
launch action doesn't work when combining --type=background with any stdin source

### ISSUE THREAD

**Issue by soystemd** [kovidgoyal/kitty#3610]
**Describe the bug**
The "launch" action doesn't launch the intended program, when
combining the "--type=background" and the "--stdin-source=@..."
options.
It works if there is no stdin-source specified.
It also works if stdin-source is @selection, but if any text is selected, it won't work.

I wanted to use this feature with a script that finds links
from the scrollback buffer and opens them.

**To Reproduce**
1. define a launch action in kitty.conf (I put firefox as an example):
`map alt+x --stdin-source=@screen_scrollback --type=background firefox`
2. save the config
3. open a new kitty instance
4. press alt+x
5. nothing happens.

**Expected behavior**
firefox (or any specified program) should be launched, with the 
specified source passed to it's standard input.

**Minimal kitty.conf**
`map alt+x launch --stdin-source=@screen_scrollback --type=background firefox`

**Environment details**
Artix Linux, X11

```
$ kitty --debug-config

kitty 0.20.3 created by Kovid Goyal
Linux desktop 5.12.2-artix1-1 #1 SMP PREEMPT Sat, 08 May 2021 13:47:18 +0000 x86_64
Loaded config files: /home/koonix/.config/kitty/kitty.conf
Running under: X11

Config options different from defaults:
update_check_interval 24.0
Added shortcuts:
	alt+x KeyAction(func='launch', args=['--stdin-source=@screen_scrollback', '--type=background', 'firefox'])
```

**Comment by trygveaa** [comment]
Duplicate of #3602, fixed in master by PR #3603.

**Comment by kovidgoyal** [comment]
dup of #3603
